static/scripts/roulette.js

```diff
@@ -1,37 +1,81 @@
 let spinButton = document.getElementById("spin");
-let redButton = document.getElementById("red");
-let blackButton = document.getElementById("black");
 let wheel = document.getElementById("wheel_img");
+let ball = document.getElementById("ball_img");
 
 const numberToPositionIndex = [0, 26, 3, 35, 12, 28, 7, 29, 18, 22, 9, 31, 14, 20, 1, 33, 16, 24, 5, 10, 23, 8, 30, 11, 36, 13, 27, 6, 37, 34, 17, 25, 2, 21, 4, 19, 15, 32]
 
 let choice = "red"
 
 async function spin() {
-    wheel.classList.add("spinning");
+    spinButton.disabled = true;
+    wheel.style.setProperty("--start-deg", "0deg");
+    wheel.style.setProperty("--target-wait-deg", "360deg");
     const timeoutPromise = new Promise(resolve => setTimeout(resolve, 2000));
     const fetchPromise = fetch("/roulette/spin", {
         method: "POST",
         headers: {'Content-Type': 'application/json'},
         body: JSON.stringify({
             "bet": document.getElementById("bet").value,
-            "choice": choice
+            "type": document.getElementById("bet-type").value,
+            "choice": document.querySelector('input[name="choice"]:checked') ? document.querySelector('input[name="choice"]:checked').value : null,
+            "number": document.getElementById("number") ? document.getElementById("number").value : null
         })
+        // "choice": choice
     }).then(response => response.text());
 
     Promise.all([fetchPromise, timeoutPromise]).then(async (values) => {
         const positionIndex = parseInt(values[0]);
         const result = numberToPositionIndex[positionIndex];
-        wheel.classList.remove("spinning");
-        wheel.style.transform = `rotate(${result * 9.473684210526316}deg)`;
-        updateUserdata();
+        const targetDeg = result * 9.473684210526316;
+        wheel.style.setProperty("--target-deg", `${targetDeg + (3 * 360)}deg`);
+        wheel.classList.add("spinning");
+        wheel.addEventListener("animationend", function () {
+            wheel.style.transform = `rotate(${result * 9.473684210526316}deg)`;
+            wheel.classList.remove("spinning");
+            wheel.style.setProperty("--start-deg", `${targetDeg}deg`);
+            wheel.style.setProperty("--target-wait-deg", `${targetDeg + 360}deg`);
+            updateUserdata();
+            spinButton.disabled = false;
+        }, {once: true});
     });
 }
 
-redButton.addEventListener("click", function () {
-    choice = "red"
-})
-blackButton.addEventListener("click", function () {
-    choice = "black"
-})
-spinButton.addEventListener('click', spin);
\ No newline at end of file
+spinButton.addEventListener('click', spin);
+
+function updateInputContainer() {
+    var selectedType = document.getElementById('bet-type').value;
+    var inputContainer = document.getElementById('input-container');
+
+    inputContainer.innerHTML = ''; // Vorherige Eingaben löschen
+
+    if (selectedType === 'color') {
```

Comment:
Schau mal das "hidden" Attribut nach. Vielleicht ist das hier eher angebracht.

Reply:
In wie fern?

Reply:
Anstatt immer das HTML immer einzufügen und rauszulöschen bei den verschiedenen Optionen, das [hidden-Attribut](https://www.w3schools.com/tags/att_hidden.asp) verwenden.

Reply:
Das einzigste ist die Radio buttons von color und evenOdd sind die gleiche Klasse wenn keiner checked ist von dem aktuellen typ also man auf fabe setzen will aber keine Farbe gesetzt hat, dann sagt der server man hat verloren, ist aber durch js nicht möglich, soll ich das aber server seitig trotzdem noch verbieten?